### PR TITLE
feat: support download uploaded documents

### PR DESCRIPTION
**Pull request by Rutheniumlmw**
part of #466

### CODE REVIEW

File: backend/app/api/routes/download.py
Comment (634750802) [inline backend/app/api/routes/download.py:1]: rename to `document.py`

File: docker-compose.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ services:
     depends_on:
       - redis
     ports:
-      - "8000:80"
+      - "8002:80"
```
Comment (634750802) [inline docker-compose.yml:17]: Do not commit irrelative configurations.

File: backend/app/api/main.py [hunk 2]
```diff
@@ -52,6 +53,7 @@
 api_router.include_router(feedback.router, tags=["chat"])
 api_router.include_router(user.router, tags=["user"])
 api_router.include_router(api_key.router, tags=["auth"])
+api_router.include_router(download.router, tags=["download_file"])
```
Comment (634750802) [inline backend/app/api/main.py:56]: api_router.include_router(document.router, tags=["documents"])

File: backend/app/api/routes/download.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+@router.get("/documents/{doc_id}/download")
+def download_file(
+    doc_id: int,
+    session: SessionDep
+):
+    doc = session.get(Document, doc_id)
+    if not doc:
+        raise HTTPException(status_code = 404, detail = "File not found")
+    
+    DATA_PATH = "../data"
```
Comment (634750802) [inline backend/app/api/routes/download.py:21]: Read the data path value from: https://github.com/pingcap/autoflow/blob/db1471d48ddb3fe9a870c01fa7e22a853054db60/backend/app/file_storage/local.py#L8

**Review by 634750802 — CHANGES_REQUESTED** [review]

File: backend/app/api/routes/download.py (hunk 3 above)
Comment (Rutheniumlmw) [reply backend/app/api/routes/download.py]: doned

File: backend/app/api/routes/document.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    doc = document_repo.must_get(session, doc_id)
+
+    name = doc.source_uri
+    filestorage = get_file_storage()
+    if filestorage.exists(name):
+        file_size = filestorage.size(name)
+        headers = {"Content-Length": str(file_size)}
+        def iterfile():
+            with filestorage.open(name) as f:
+                while chunk := f.read(8192):  # 每次读取 8KB
```
Comment (634750802) [inline backend/app/api/routes/document.py:24]: Please refer to [FastAPI Document](https://fastapi.tiangolo.com/advanced/custom-response/#using-streamingresponse-with-file-like-objects). It's not necessary to manually call `f.read()`.

Comment (Rutheniumlmw) [reply backend/app/api/routes/document.py]: done

**Review by 634750802 — APPROVED** [review]
nice work

**Review by Mini256 — APPROVED** [review]

**Pull request merged**